src/PhpSpec/CodeGenerator/Generator/PromptingGenerator.php

```diff
@@ -39,20 +41,29 @@
      * @var \PhpSpec\Util\Filesystem
      */
     private $filesystem;
+
     /**
      * @var ExecutionContextInterface
      */
     private $executionContext;
 
     /**
-     * @param IO               $io
+     * @var EventDispatcherInterface
+     */
+    private $dispatcher;
+
+    /**
+     * @param IO $io
      * @param TemplateRenderer $templates
-     * @param Filesystem       $filesystem
+     * @param EventDispatcherInterface $dispatcher
+     * @param Filesystem $filesystem
+     * @param ExecutionContextInterface $executionContext
      */
-    public function __construct(IO $io, TemplateRenderer $templates, Filesystem $filesystem = null, ExecutionContextInterface $executionContext = null)
+    public function __construct(IO $io, TemplateRenderer $templates, EventDispatcherInterface $dispatcher, Filesystem $filesystem = null, ExecutionContextInterface $executionContext = null)
```

Comment:
OR actually this takes the total dependency count quite high, maybe it could be achieved via Decoration instead (just an idea)?